### ISSUE TITLE
No feedback when trying to create note w/ empty comment but attached image

### ISSUE THREAD

**Issue by rugk** [westnordost/StreetComplete#1340]
**STR:**
1. Create a new note. (click on that button)
2. Do not enter any text, but add an image.

What happens: I cannot press "OK". Also, I get no feedback when i press "OK" that it does not work, just nothing happens.

What should happen: IMHO, allow me to continue with empty comments when I attach an image.
What should happen: Also, probably some error feedback should be there, in case, I try to submit a comment i cannot submit yet.

**Comment by ENT8R** [comment]
I don't think that this would be an improvement because without any comment, other mappers resolving the note(s) probably don't know exactly what the user wants to tell them... At least a small comment should be added to every note IMO so that other mappers can resolve the note as fast as possible, hopefully without asking questions to the original creator of the note.
If there is just a picture of e.g. a bus stop or a building attached to the note, I, as a mapper resolving notes, would not know at all what I should edit/improve then...

**Comment by rugk** [comment]
But opening hours or so – in combination with the StreetComplete question you have – are likely very easy to interpret. If you e.g. take a picture of the opening hours sign…

**Comment by ENT8R** [comment]
I agree, in combination with the question it might be possible for the mapper to infer what the creator of the note wants to express. But I thought you were only talking about the case when a user of SC creates a new note unrelated to any question. In that case still think that it might be sometimes hard to figure out what the user wants to express.

**Comment by westnordost** [comment]
Also in the other case. StreetComplete only adds "Can't answer XYZ: ... ". What kind of answer or description is that, just adding a few photos?

rugk, you should describe your specific use case you were confronted with.

**Comment by rugk** [comment]
…as already explained, e.g. for opening hours a picture of the sign 100%ly answers he question…

**Comment by rugk** [comment]
I just don't think it makes sense to limit the user there. If the user is too lazy to add a proper text, in case it is needed, it's not the problem of the app. Actually, there are cases, where an additional text makes little sense.

**Comment by westnordost** [comment]
Are you speaking of yourself?

**Comment by rugk** [comment]
No, for the universe... 😜 

At least, I can say,

**Comment by westnordost** [comment]
My point is, that I would prefer to know what *your* specific problem with the UI is, not what "a user" might want or not want to do. This is probably a misunderstanding and you meant yourself and specifically the opening hours quest all along.

So, my answer to this: It is still at least impolite to just post a picture of the opening hours sign without giving a comment like "too complicated", "can't select this or that" or "no time right now to input it on the go, here is a photo". The app does not and shall not encourage to just create notes just because it may be quicker than actually answering the question, even in case the latter is possible. This is why at least half-a-sentence of an excuse or proper reason to create the note should be added.

However, it is not good that the OK-button appears clickable but is not and there is also no feedback when clicking it. This should be changed.

**Comment by rugk** [comment]
Okay, so actually you are right: IIRC when I noticed this issue, I was solving an opening hours quest. So feel free to think specific about that.
However, i thought and think it is not really useful to allow empty comments only for one quest type and I still think it makes sense for other quest types, too. (maybe street name or so, ...)

Reg. impolite: maybe, yeah. But brevity is king... 😉 And a picture says more than thousand words, so it is likely I do not need to explain what is shown in the picture again...
Also, it hardly matters for the ones solving the note with a picture of an opening hours sign, whether I just had no time to enter the data or it was too complicated. Possibly it matters for SC devs/you here, but yeah...

> "no time right now to input it on the go, here is a photo"

I guess this was the reason in my case. It was possibly some complicated sign and I had no time. (I also usually try to solve the notes I created with SC myself later with a proper editor like JOSM or so, so it's not I want to unload work to others.) And obviously it also is not a fast way to write this long text as a note everytime. If you already have no time, you hardly have time to write that note. (Admittingly, https://github.com/westnordost/StreetComplete/issues/1191 would also solve the time issue here, so that can be an alternative.)
